src/SWP/TemplateEngineBundle/Controller/ContainerController.php

```diff
@@ -0,0 +1,280 @@
+     * )
+     * @Route("/api/{version}/templates/containers/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_update_container")
+     * @Method("PATCH")
+     */
+    public function updateAction(Request $request, $id)
+    {
+        if (!$id || !is_numeric($id)) {
+            throw new UnprocessableEntityHttpException('You need to provide container Id (integer).');
+        }
+
+        $entityManager = $this->get('doctrine')->getManager();
+        $container = $entityManager->getRepository('SWP\TemplateEngineBundle\Model\Container')->getById($id)->getOneOrNullResult();
+
+        if (!$container) {
+            throw new NotFoundHttpException('Container with this id was not found.');
+        }
+
+        $form = $this->createForm(ContainerType::class, $container, array(
+            'method' => $request->getMethod(),
+        ));
+
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            $extraData = $form->get('data')->getExtraData();
+            if ($extraData && is_array($extraData)) {
+                // Remove old containerData's
+                foreach ($container->getData() as $containerData) {
+                    $entityManager->remove($containerData);
+                }
+
+                // Apply new containerData's
+                foreach ($extraData as $key => $value) {
+                    $containerData = new ContainerData($key, $value);
+                    $containerData->setContainer($container);
+                    $entityManager->persist($containerData);
+                    $container->addData($containerData);
+                }
+            }
+
+            $entityManager->flush($container);
+            $entityManager->refresh($container);
+
+            return $this->handleView(View::create($container, 201));
+        }
+
+        return $this->handleView(View::create($form, 200));
+    }
+
+    /**
+     * Link or Unlink resource with Container.
+     *
+     * **link or unlink widget**:
+     *
+     *     header name: "link"
+     *     header value: "</api/{version}/templates/widgets/{id}; rel="widget">"
+     *
+     * or with specific position:
+     *
+     *     header name: "link"
+     *     header value: "</api/{version}/templates/widgets/{id}; rel="widget">,<1; rel="widget-position">"
+     *
+     * @ApiDoc(
+     *     statusCodes={
+     *         201="Returned when successful",
+     *         404="Returned when resource not found",
+     *         409={
+     *           "Returned when the link already exists",
+     *         }
+     *     }
+     * )
+     *
+     * @Route("/api/{version}/templates/containers/{id}", defaults={"version"="v1"}, name="swp_api_templates_link_container")
+     *
+     * @Method("LINK|UNLINK")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function linkUnlinkToContainerAction(Request $request, $id)
```

Comment:
Lines from 205 to 245 are unique for that function, i can create some service and pass content of lines 205-245 as anonymous function to be called in service - what do you think?